packages/react/src/components/Loader/Loader.tsx

```diff
@@ -51,27 +61,61 @@ class Loader extends UIComponent<ReactProps<LoaderProps>> {
       content: false,
       color: true,
     }),
+    delay: PropTypes.number,
     indicator: customPropTypes.itemShorthand,
+    inline: PropTypes.bool,
     label: customPropTypes.itemShorthand,
     labelPosition: PropTypes.oneOf(['above', 'below', 'start', 'end']),
     size: customPropTypes.size,
   }
 
   static defaultProps = {
     accessibility: loaderBehavior,
+    delay: 0,
     indicator: '',
     labelPosition: 'below',
     size: 'medium',
   }
 
+  delayTimer: number
+
+  constructor(props, context) {
+    super(props, context)
+
+    this.state = {
+      visible: this.props.delay === 0,
+    }
+  }
+
+  componentDidMount() {
+    const { delay } = this.props
+
+    if (delay > 0) {
+      this.delayTimer = window.setTimeout(() => {
+        this.setState({ visible: true })
```

Comment:
Seems like we should clear the timeout here too and set this.delayTimer = null

Reply:
They will never collide because it's called in `componentDidMount()`, there are no obvious reasons to clear it.